builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -143,7 +150,14 @@ func resourceAwsS3BucketObjectPut(d *schema.ResourceData, meta interface{}) erro
 		putInput.ContentDisposition = aws.String(v.(string))
 	}
 
-	resp, err := s3conn.PutObject(putInput)
```

Comment:
Good catch. I missed setting ServerSideEncryption instead of setting the header directly.